Added new option to clicking on the icon of a running application

ihsankavalci opened the pull request:
If running application has more than 1 window don't close the window preview and do nothing. If running application has 1 window minimize or activate. This behaviour is more similar to Windows.

charlesg99 commented on appIcons.js:
```diff
@@ -753,7 +750,11 @@ var taskbarAppIcon = new Lang.Class({
         }
-        
+
+        let appCount = this.getAppIconInterestingWindows().length;
```
Please modify the `appIsRunning` assignation to use the new `appCount` variable, so we don't call `this.getAppIconInterestingWindows()` twice in a row

ihsankavalci replied:
I did it, thank you.

ihsankavalci commented:
Thank you for feedbacks.
- Actually, I couldn't find a proper name for this. Your suggestion seems reasonable to me. I have changed namings.
- I did not fully understand what (!Main.overview._shown) was doing, but i fixed this to the more correct way I think.
The only missing side from Windows is; If there are multiple windows open, it can't close/open window previews. But I can do it in future.

jderose9 commented:
I made a few changes.

- When I clicked the a single-instance application it would open the window, when I clicked again, it would remain open, and not until 3rd click would it minimize. So I fixed this
- The existing "Minimize" option really has two behavior depending on whether it is the primary click action or a modifier/middle button action. When it is primary it is actually a toggle and when secondary it minimizes. So I've relabelled this to Toggle in the primary click action dropdown.
- Handled the ungrouped applications scenario.
- Added Toggle + Preview to the secondary click action dropdowns

Let me know what you think. Thanks.

Also, let's keep this one out until v18 is tagged since it will break the translations.
